llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -528,7 +528,10 @@ bool MachineSinking::PerformSinkAndFold(MachineInstr &MI,
           continue;
         MachineInstr *NewDbgMI = SinkDst->getMF()->CloneMachineInstr(DbgMI);
         SinkMBB.insertAfter(InsertPt, NewDbgMI);
-        NewDbgMI->getOperand(0).setReg(DstReg);
```

Comment:
To clarify, that `getOperand(0)` works for `DBG_VALUE`s only, other debug instructions can have operands using `DefReg` at other positions.